testing-commandments.md

```diff
@@ -0,0 +1,30 @@
+# 8 Commandments of Automated Testing
+
+*DISCLAIMER* - Failure to meet any of these points doesn't mean new ideas or pull requests should be outright rejected. They should be used as points of discussion to ultimately help us ship code faster.
+
+`code` is used in reference to the code being tested. The target, maybe?
+`test` is used in reference to the code that is exercising the codebase to validate correctness.
+
+## Thou shalt be consistent within projects
+If a project has a de-facto standard, it should be maintained. The process for changing a de-facto standard should be people focused, and should not be initially communicated via a pull request.
+
+## Thy tests should be obvious
+If the test behaviour isn't obvious by looking at it, it's no good. If I have to scroll for days to read a test, it's no good.
+
+## Thy tests should only check for something once
+If I make a single code change and multiple tests fail, it's no good.
+
+## Broken code should break thy tests
+If I can delete code and it's corresponding test suite still passes, it's no good.
```

Comment:
I would argue that 'deleting' is not the key point here, ANY changes to the code that alter the interface in a way that means it no longer meets the specification it is expected to match should break the tests.